### PR TITLE
Add node watcher to testnet scripts

### PR DESCRIPTION
**Pull request by camilbancioiu**
There is now a variable called `NODE_WATCHER`, set by default to 1. This variable enables the so-called node watcher, a daemon which restarts the node if it dies.

Enabling the node watcher will also make the node a daemon. This means that it cannot be stopped with CTRL+C from tmux anymore. 

A better solution must be implemented. The current node watcher was designed with a different use case in mind, and is somewhat incompatible with the testnet scripts.

### CODE REVIEW

File: scripts/testnet/include/validators.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+resumeValidators() {
+  iterateOverValidators resumeSingleValidator
+}
+
+stopValidators() {
+  iterateOverValidators stopSingleValidator
+}
+
+iterateOverValidators() {
```
Comment (iulianpascalau) [inline scripts/testnet/include/validators.sh:22]: almost duplicate code with iterateOverObservers

File: scripts/testnet/include/validators.sh [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  local source_command="source $ELRONDTESTNETSCRIPTSDIR/include/watcher.sh"
+  local watcher_command="node-start-with-watcher $VALIDATOR_INDEX $PORT &"
+  local node_command=$(assembleCommand_startValidatorNode $VALIDATOR_INDEX)
+  echo "$node_command" > $WORKING_DIR/node-command
+  echo "$PORT" > $WORKING_DIR/node-port
+
+  echo "$source_command ; $watcher_command"
+}
+
+assembleCommand_startValidatorNode() {
```
Comment (iulianpascalau) [inline scripts/testnet/include/validators.sh:105]: this doesn't actually start the node but assemble the command line for a node

File: scripts/testnet/include/validators.sh (hunk 1 above)
Comment (sasurobert) [reply scripts/testnet/include/validators.sh:22]: +1

File: scripts/testnet/include/validators.sh (hunk 2 above)
Comment (sasurobert) [reply scripts/testnet/include/validators.sh:105]: +1

**Review by sasurobert — CHANGES_REQUESTED** [review]
remove duplicated code

File: scripts/testnet/include/validators.sh (hunk 2 above)
Comment (camilbancioiu) [reply scripts/testnet/include/validators.sh:105]: Yes, the function is called `assembleCommand_startValidatorNode()`. Can rename to something else if unclear.

Comment (iulianpascalau) [reply scripts/testnet/include/validators.sh:105]: can remain as it is

File: scripts/testnet/include/validators.sh (hunk 1 above)
Comment (iulianpascalau) [reply scripts/testnet/include/validators.sh:22]: can remain as it is

**Review by iulianpascalau — APPROVED** [review]

File: scripts/testnet/include/validators.sh (hunk 1 above)
Comment (camilbancioiu) [reply scripts/testnet/include/validators.sh:22]: Due to Bash specifics, these two duplicated functions are cleaner than a single deduplicated iterator (at least at my level of knowledge). A future rewrite in Python will allow for cleaner code.

**Review by sasurobert — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**